src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -4458,21 +4458,33 @@ protected function addCrossFkScheduledForDeletion(&$script, CrossForeignKeys $cr
             foreach ($crossFKs->getIncomingForeignKey()->getColumnObjectsMapping() as $reference) {
                 $local   = $reference['local'];
                 $foreign = $reference['foreign'];
+                $value   = $reference['value'];
 
                 $idx = array_search($local, $crossPks, true);
-                $script .= "
+                if ($value) {
+                    $script .= "
+                        \$entryPk[$idx] = '$value';";
+                } else {
+                    $script .= "
                         \$entryPk[$idx] = \$this->get{$foreign->getPhpName()}();";
+                }
             }
 
             $combinationIdx = 0;
             foreach ($crossFKs->getCrossForeignKeys() as $crossFK) {
                 foreach ($crossFK->getColumnObjectsMapping() as $reference) {
                     $local   = $reference['local'];
                     $foreign = $reference['foreign'];
+                    $value   = $reference['value'];
 
                     $idx = array_search($local, $crossPks, true);
-                    $script .= "
+                    if ($value) {
+                        $script .= "
+                        \$entryPk[$idx] = '$value';";
+                    } else {
+                        $script .= "
                         \$entryPk[$idx] = \$combination[$combinationIdx]->get{$foreign->getPhpName()}();";
+                    }
```

Comment:
Because `addCrossFkScheduledForDeletion` is already quite complex we should do our best to keep reducing it whenever possible. I see an opportunity here to simply put this additional logic into a protected method.

It could also help reduce duplication, as the other two code hunks in this patch look very similar to me.